### PR TITLE
[WIP] Loot System

### PR DESCRIPTION
**Pull request by chemicstry**
Creating this PR for the loot system update. Currently all loot is tied to an entity (Loot class as an entity member) but blizz started using HighGuid::LootObject and all loot packets come with it. So it would make sense to split all loot into it's own container and lookup by loot guid. Also current loot system does not support currencies.

This commit enables basic looting functions and it could be merged as is but it may get complicated with this old system in the future. I hope to get some input on this.

### CODE REVIEW

File: src/server/game/Handlers/LootHandler.cpp [hunk 1]
```diff
@@ -33,71 +33,74 @@
...
-    else if (lguid.IsCorpse())
-    {
-        Corpse* bones = ObjectAccessor::GetCorpse(*player, lguid);
-        if (!bones)
+        else if (lguid.IsItem())
         {
-            player->SendLootRelease(lguid);
-            return;
-        }
+            Item* pItem = player->GetItemByGuid(lguid);
```
Comment (DDuarte) [inline src/server/game/Handlers/LootHandler.cpp]: might as well remove hungarian notation while changing this :P

**Comment by msoky** [comment]
any news about this important function? after this, players can do quests with loot requirements

**Comment by Intel** [comment]
Looting with this PR is already working if you really want to use it. However, I'm still working on AoE looting which will come with a little rewrite of looting system.

**Comment by msoky** [comment]
AoE looting is looting from multiple corpses around player at once? :)

File: src/server/game/Loot/LootMgr.h [hunk 2]
```diff
@@ -29,6 +29,19 @@
...
+namespace WorldPackets
+{
+    namespace Loot
+    {
+        class LootResponse;
+    }
+
+    namespace Item
+    {
+        class ItemInstance;
```
Comment (cemak) [inline src/server/game/Loot/LootMgr.h]: warning C4099

**Comment by DDuarte** [comment]
@Intel: I suggest pushing this to the repo as is to prevent the PR from stalling.
If there's anything that needs to be changed it can be changed later.

**Comment by msoky** [comment]
there is some warnings during compile in VS2013 but its working in game...

**Comment by Intel** [comment]
@DDuarte Ok, I'll push it to core and implement AoE looting part when it's done. I was quite busy for the past weeks that's why it stalled.

**Comment by msoky** [comment]
Any news about new loot system?

**Comment by Intel** [comment]
It's currently half done, but I was on vacation so that's why it is taking so long.

**Comment by msoky** [comment]
any news after month? :)

**Comment by msoky** [comment]
any news after month? :)

**Comment by msoky** [comment]
any news about AOE looting?

**Pull request closed without merging**